plugins/modules/managed_systems_facts.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/python
```

Comment:
The point of this new module is to gather facts from an HMC target as a whole. The case I was thinking of, for example, was when the user needed a way to automate multiple updates for a specific type of system being managed by their fleet of HMCs. This allows them to query for all managed systems (through lssysconn) in the specific inventory item (HMC) and get the entire list, which they can then filter for certain Machine Types and do all updates without having to specify all the different MTMS from a hard list that they have to filter manually. Relieving the user of that burden seems beneficial, and it also removes the need to call other HMC commands to ensure a System is acted on by the HMC that is managing it, so it's a more efficient way (speed/load wise) to handle it.

I'm not sure if the power_system could be used the same way, but I hadn't noticed it before. Do let me know if there's a way to do the above with this one, or if we can move functionality from the new module to this one. The latter didn't seem obvious because power_system seems to be used to target specific systems, so separating them made sense to keep with the Ansible doc's suggestion of "doing one thing well".
Basically, the intent is to use it for facts on the HMC about the systems being managed, while the power_system module would be used for action/facts about one system specifically.

> we do have a plan to include servers info into the inventory list using inventory plugin (customer request), pipelined for Dec release
This one might work though I'm not sure I understand how it works yet, I'll look into it. As long as the user gets a list of the managed systems of the inventory item in that iteration, and it can be used with the yaml filters to pick out a specific machine type as part of another play in the same playbook it should be the same thing (albeit probably slower due to the extra overhead of the inventory module).

Reply:
Yes i understood the use case, and it make sense as well. But as I mentioned on earlier thread, we have a customer request to include CEC/Server info into the existing inventory plugin module (still need to evaluate the feasibility of it) and if that happens then using CEC group created by inventory plugin, user should be able to iterate and do firmware update.

Reply:
> and if that happens then using CEC group created by inventory plugin, user should be able to iterate and do firmware update.

Sounds good. Module removed.